pkg/gc/gc.go

```diff
@@ -69,12 +69,14 @@ func Tricolor(roots []Node, refs func(ref Node) ([]Node, error)) (map[Node]struc
 	)
 
 	grays = append(grays, roots...)
+	for _, root := range roots {
+		seen[root] = struct{}{} // pre-mark this as not-white
```

Comment:
we can probably append root one by one as well and check existence before appending to avoid duplications from `roots`. But I think it's unlikely we have duplications from `roots` and some duplications won't impact much.

Reply:
Duplicated roots is not a concern